tests/lib/rules/newline-per-chained-call.js

```diff
@@ -163,34 +167,304 @@ ruleTester.run("newline-per-chained-call", rule, {
         options: [{ ignoreChainWithDepth: 1 }],
-        errors: [{ messageId: "expected", data: { callee: "[biz]" } }]
+        errors: [{ messageId: "expectedLineBreak", data: { propertyName: "[biz]" } }]
     }, {
         code: "(foo).bar().biz()",
         output: "(foo).bar()\n.biz()",
         options: [{ ignoreChainWithDepth: 1 }],
-        errors: [{ messageId: "expected", data: { callee: ".biz" } }]
+        errors: [{ messageId: "expectedLineBreak", data: { propertyName: ".biz" } }]
     }, {
         code: "foo.bar(). /* comment */ biz()",
         output: "foo.bar()\n. /* comment */ biz()",
         options: [{ ignoreChainWithDepth: 1 }],
-        errors: [{ messageId: "expected", data: { callee: ".biz" } }]
+        errors: [{ messageId: "expectedLineBreak", data: { propertyName: ".biz" } }]
     }, {
         code: "foo.bar() /* comment */ .biz()",
         output: "foo.bar() /* comment */ \n.biz()",
         options: [{ ignoreChainWithDepth: 1 }],
-        errors: [{ messageId: "expected", data: { callee: ".biz" } }]
+        errors: [{ messageId: "expectedLineBreak", data: { propertyName: ".biz" } }]
+    },
+
+    /*
+     * maxTotalChainDepth: 1
+     */
+    {
+        code: "_\n.chain({}).map(foo).filter(bar).value();",
+        output: "_\n.chain({})\n.map(foo)\n.filter(bar)\n.value();",
+        options: [{ maxTotalChainDepth: 1 }],
+        errors: [{
+            messageId: "expectedLineBreak", data: { propertyName: ".map" }
+        }, {
+            messageId: "expectedLineBreak", data: { propertyName: ".filter" }
+        }, {
+            messageId: "expectedLineBreak", data: { propertyName: ".value" }
+        }]
+    }, {
+        code: "_\n.chain({})\n.map(foo)\n.filter(bar).value();",
+        output: "_\n.chain({})\n.map(foo)\n.filter(bar)\n.value();",
+        options: [{ maxTotalChainDepth: 1 }],
+        errors: [{
+            messageId: "expectedLineBreak", data: { propertyName: ".value" }
+        }]
+    }, {
+        code: "a().b().c().e.d()",
+        output: "a()\n.b()\n.c()\n.e\n.d()",
+        options: [{ maxTotalChainDepth: 1 }],
+        errors: [{
+            messageId: "expectedLineBreak", data: { propertyName: ".b" }
+        }, {
+            messageId: "expectedLineBreak", data: { propertyName: ".c" }
+        }, {
+            messageId: "expectedLineBreak", data: { propertyName: ".e" }
+        }, {
+            messageId: "expectedLineBreak", data: { propertyName: ".d" }
+        }]
+    }, {
+        code: "a.b.c().e().d()",
+        output: "a\n.b\n.c()\n.e()\n.d()",
+        options: [{ maxTotalChainDepth: 1 }],
+        errors: [{
+            messageId: "expectedLineBreak", data: { propertyName: ".b" }
+        }, {
+            messageId: "expectedLineBreak", data: { propertyName: ".c" }
+        }, {
+            messageId: "expectedLineBreak", data: { propertyName: ".e" }
+        }, {
+            messageId: "expectedLineBreak", data: { propertyName: ".d" }
+        }]
+    }, {
```

Comment:
The `a\n` part seems to be out of scope of the original issue #12970, so it becomes difficult to understand what's the actual proposal.

The purpose of PR #5365 (issue #5289) was exactly to ignore "not a method call" property access such as `a.b`, because it was a bug. The same PR may have accidentally introduced some other changes in the behavior, and I think we should focus on that for the start.

Also, as you noted in the description of this PR, this new option changes several different things in the behavior of this rule and basically turns it into another rule. Would it be possible to split this into several options (e.g. one to enforce linebreaks for all elements, the other to treat property access as a part of the chain) instead?

Reply:
I agree, adding back in `MemberExpression` is not part of the original issue's example, but it was actually my goal the entire time. I didn't realize this functionality change as all the examples used function calls.

If we renamed this to `newline-per-chained-member`, then I could add 2 props:
- `includeMethodCalls` (default `true`)
- `includeNonMethodCalls` (default `true`)

The reason I'd default both to `true` is only if we renamed. If we did _not_ rename, then I'd default `includeNonMethodCalls` to `false`. For now, I'll make this change without renaming and will use the `false` value to retain the original functionality.